compute_field_after_install/models/recompute_field.py

```diff
@@ -0,0 +1,137 @@
+from odoo.tools import config
+from odoo.tools.translate import _
+
+_logger = logging.getLogger(__name__)
+
+
+class RecomputeField(models.Model):
+    _name = "recompute.field"
+    _description = "Recompute Field"
+
+    model = fields.Char(required=True)
+    field = fields.Char(required=True)
+    last_id = fields.Integer(
+        help="Last record ID on which computing have been executed"
+    )
+
+    step = fields.Integer(
+        required=True,
+        help="Recomputing batch size.",
+        compute="_compute_default_step",
+        store=True,
+        readonly=False,
+        precompute=True,
+    )
+    state = fields.Selection(
+        [
+            ("todo", "Todo"),
+            ("done", "Done"),
+        ]
+    )
+
+    @api.depends("model", "field")
+    def _compute_default_step(self):
+        for recompute_field in self:
+            model = recompute_field.model.replace(".", "_")
+            recompute_field.step = config.get(
+                f"computed_fields_batch_size__{model}__{recompute_field.field}",
+                config.get(
+                    f"computed_fields_batch_size__{model}",
+                    config.get("computed_fields_batch_size", 1000),
+                ),
+            )
+
+    @api.constrains("step")
+    def _check_step(self):
+        for recompute_field in self:
+            if recompute_field.step <= 0:
+                raise UserError(_("Step must be greater than 0"))
+
+    @api.model
+    def _run_all(self):
+        return self.search([("state", "=", "todo")]).run()
+
+    def run(self):
+        # Group tasks by compute method to avoid computing multifields computes multiple times
+
+        def group_key(task):
+            return task.model, self.env[task.model]._fields[task.field].compute
+
+        model_tasks = groupby(
+            self.sorted(key=group_key),
+            group_key,
+        )
+        for (model, _compute_fun), tasks in model_tasks:
+            tasks = reduce(lambda x, y: x | y, tasks)
+            fields = set(tasks.mapped("field"))
+            last_id = max(tasks.mapped("last_id"), default=None)
+            step = min(tasks.mapped("step"))
+
+            while True:
+                _logger.info(
+                    "Recompute fields %s for model %s in background. Last id %d",
+                    fields,
+                    model,
+                    last_id,
+                )
+                records = self.env[model].search(
+                    [("id", "<", last_id)] if last_id else [],
+                    limit=step,
+                    order="id desc",
+                )
+                if not records:
+                    tasks.state = "done"
+                    self.env.cr.commit()  # pylint: disable=E8102
+                    break
+                for field in fields:
+                    field_ = records._fields[field]
+                    self.env.add_to_compute(field_, records)
+
+                records.recompute()
```

Comment:
Functionnaly working but with a depreciation warning on recompute():
`compute_field_after_install/models/recompute_field.py:101: DeprecationWarning: Deprecated method recompute(), use flush_model(), flush_recordset() or env.flush_all() instead`
```suggestion
                self.env.flush_recordset(records)
```